compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -672,6 +671,20 @@ object Scanners {
         reset()
         if atEOL then token = COLONeol
 
+    // consume => and insert <indent> if applicable
+    def observeArrowIndented(): Unit =
+      if isArrow && indentSyntax then
+        peekAhead()
+        val atEOL = isAfterLineEnd || token == EOF
+        reset()
+        if atEOL then
+          val nextWidth = indentWidth(next.offset)
+          val lastWidth = currentRegion.indentWidth
+          if lastWidth < nextWidth then
+            currentRegion = Indented(nextWidth, COLONeol, currentRegion)
```

Comment:
Is it expected to use `COLONeol` here? Or could use `ARROW` instead?

At first sight, it only seems to be used [here](https://github.com/scala/scala3/blob/b57549e66fb3e63f18adc674fe1ae4a5fd4c1fa9/compiler/src/dotty/tools/dotc/parsing/Scanners.scala#L704) where it's compared to `MATCH`.

Reply:
My first attempt used ARROW and used it during newline handling, but that was not needed. I changed to colon for uniformity only; the result here is only produced and discarded in lookahead scanner. Actually on my WIP branch, I used a new token `ARROWeol` for when it is not a proper indent, as after `MATCH`.

Reply:
Okay, let's get that merged then!